simulation/chart.py

```diff
@@ -1,232 +1,427 @@
 # Copyright 2022-, Semiotic AI, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
+from enum import Enum
+from functools import singledispatch
 from subprocess import Popen
-from typing import Optional, Tuple
+from typing import Any, Dict, Literal, NamedTuple, Optional, Tuple, TypedDict, Union
 
 import ffmpeg
 import pyqtgraph as pg
 from pyqtgraph.Qt import QtCore, QtGui
 
 
-def create_layout(
-    size: Tuple[int, int],
-    title: str,
-    show: bool,
-    antialias: Optional[bool] = None,
-    foreground=None,
-) -> pg.GraphicsLayoutWidget:
-    """Initialize application and layout container
+class RGBAColor(NamedTuple):
+    """Color model based on red, green, blue and an alpha/opacity value.
 
-    Args:
-        size (Tuple[int,int]):(width,height) of the charts window.
-        title (str): Title of charts window.
-        show (bool): Show app window.
-        antialias (Optional[bool], optional): Use antialiasing. If true, smooth visuals and slower refresh rate. Defaults to None.
-        foreground (Any, optional): General foreground color (text,ie). Defaults to None.
+    Attributes:
+        R (int): value between 0 and 255 for red value.
+        G (int): value between 0 and 255 for green value.
+        R (int): value between 0 and 255 for blue value.
+        A (int, optional): value between 0 and 255 for alpha/opacity value.
     """
-    # Set up PyQtGraph
-    if foreground is not None:
-        pg.setConfigOption("foreground", foreground)
-    if antialias is not None:
-        pg.setConfigOptions(antialias=antialias)
 
-    pg.mkQApp(title)
+    R: int
+    G: int
+    B: int
+    A: Optional[int] = None
```

Comment:
Is there a reasonable default value other than `None` for `A`? I think `255` is probably fine here.

Reply:
You're right pyqtgraph mkColor is replacing `None` with 255.

Reply:
https://github.com/semiotic-ai/autoagora-agents/pull/21/commits/0d488125217f7472c649d43cfd26daae40c6cbec